### PR TITLE
Add evaluate percent tick to expression

### PR DESCRIPTION
**Pull request by RoyToledano**
- Evaluate, Percent and Ticks side methods added.
- Create 'TicksCounter' object.
- Add in each action invoke the current ticks count in order to save this information. (for Ticks side method).
- Handle some todos.

### CODE REVIEW

File: Engine/src/simulation/objects/world/ticks/counter/TicksCounter.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * This class holds ticks for a simulation.
+ */
+public class TicksCounter {
+    private int ticks = 0;
+
+    public int getTicks() {
+        return ticks;
+    }
+
+    public void moveTicksByOne(){
```
Comment (Ronenii) [inline Engine/src/simulation/objects/world/ticks/counter/TicksCounter.java]: Change this to addTick or incrementTIck

File: Engine/src/simulation/objects/entity/Entity.java [hunk 2]
```diff
@@ -101,12 +101,12 @@ public int hashCode() {
      * @param action      The action to invoke on all instances.
      * @param probability The probability of this action to invoke on instances.
      */
-    public void invokeActionOnAllInstances(Action action, double probability) {
+    public void invokeActionOnAllInstances(Action action, double probability, int lastChangTickCount) {
```
Comment (Ronenii) [inline Engine/src/simulation/objects/entity/Entity.java:104]: What does lastChang mean? last change?

**Review by Ronenii — COMMENTED** [review]
I didn't quite get why we need a tick counter object and why almost everything needs to have it in some way or another?

File: Engine/src/simulation/objects/entity/Entity.java (hunk 2 above)
Comment (RoyToledano) [reply Engine/src/simulation/objects/entity/Entity.java:104]: Change* fast typing

File: Engine/src/simulation/objects/world/ticks/counter/TicksCounter.java (hunk 1 above)
Comment (RoyToledano) [reply Engine/src/simulation/objects/world/ticks/counter/TicksCounter.java]: Tovush

**Review by Ronenii — APPROVED** [review]
Approved

**Pull request merged**